python/ray/data/context.py

```diff
@@ -164,87 +125,130 @@
 
 
 @DeveloperAPI
+@dataclass
 class DataContext:
-    """Singleton for shared Dataset resources and configurations.
-
-    This object is automatically propagated to workers and can be retrieved
-    from the driver and remote workers via DataContext.get_current().
+    """Global settings for Ray Data.
+
+    Configure this class to enable advanced features and tune performance.
+
+    .. note::
+        This object is automatically propagated to workers. You can access it
+        from the driver and remote workers via :meth:`DataContext.get_current()`.
```

Comment:
also notice users to apply changes before creating their datasets, because DataContext will be sealed after a dataset is created.